paf-mvp-audit/src/view.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * Resources used by the controller for HTML views and CSS.
+ * TODO: fix the warning associated with can't find module or type.
+ */
+import logoSvg from './images/OneKey.svg';
+import css from './css/ok-ui.css';
+import auditTemplate from './views/audit.html';
+import buttonTemplate from './views/button.html';
+import { Locale } from './locale';
+import { IView } from '@core/ui/binding';
+import { Log } from '@core/log';
+
+/**
+ * Type to use with HTML views that support locale language customization.
+ */
+type ViewTemplate = (l: Locale) => string;
+
+export class View implements IView {
+  // The shadow root for the UI.
+  public root: ShadowRoot = null;
+
+  // The outer container for the UI.
+  private outerContainer: HTMLElement = null;
+
+  // The element that contains the advert. Used to add the UI components to the DOM around the advert.
+  private readonly advert: HTMLElement;
+
+  // The container element for the UI, or null if the UI has not yet been added to the DOM.
+  private auditContainer: HTMLDivElement | null = null;
+
+  // Log used to output status messages.
+  private readonly log: Log;
+
+  // The locale that the UI should adopt.
+  public readonly locale: Locale;
+
+  /**
+   * Constructs a new instance of Controller.
+   * @param advert element the module relates to
+   * @param locale the language file to use with the UI
+   */
+  constructor(advert: HTMLElement, locale: Locale, log: Log) {
```

Comment:
(for another PR) suggestion to use more concise shorthand definition of properties (https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd):

```typescript
// Remove lines 26, 32 and 35
constructor(private readonly advert: HTMLElement, public readonly locale: Locale, private readonly log: Log) {
  this.locale.Logo = logoSvg;
}
```

Reply:
Thanks. We'll consider for next PR.